### PR TITLE
feat(config): support regexes for commit triggers

### PR DESCRIPTION
**Pull request by mfocko**
Signed-off-by: Matej Focko <mfocko@redhat.com>

TODO:

- [x] Write new tests or update the old ones to cover new functionality.
- [x] Update doc-strings where appropriate.
- [ ] Update or write new documentation in `packit/packit.dev`.

---

RELEASE NOTES BEGIN
You can now specify branches for a job triggered by a commit with a regex.
RELEASE NOTES END

### CODE REVIEW

File: packit_service/worker/checker/koji.py [hunk 1]
```diff
@@ -37,7 +38,7 @@ def pre_check(self) -> bool:
             PushPagureEvent.__name__,
         ):
             configured_branch = self.koji_build_helper.job_build_branch
-            if self.data.git_ref != configured_branch:
+            if not re.match(configured_branch, self.data.git_ref):
```
Comment (TomasTomecek) [inline packit_service/worker/checker/koji.py:41]: wow, that's easy

Comment (TomasTomecek) [reply packit_service/worker/checker/koji.py:41]: Or write an exploit in regex :|

**Review by TomasTomecek — APPROVED** [review]
document this also in packit.dev please

**Comment by mfocko** [comment]
> document this also in packit.dev please

@TomasTomecek I'm actually not sure if I want to document it before I test it in action :D

**Comment by TomasTomecek** [comment]
> > document this also in packit.dev please
> 
> @TomasTomecek I'm actually not sure if I want to document it before I test it in action :D

yes, absolutely! just pointing out that once we make sure it works, we should document it :)

**Pull request merged**